muikku-core/src/main/java/fi/otavanopisto/muikku/schooldata/WorkspaceEntityController.java

```diff
@@ -255,6 +258,29 @@ public boolean isSignupAllowed(WorkspaceEntity workspaceEntity, UserGroupEntity
         if (signupGroupIdentifiersObject instanceof Collection) {
           Collection<?> signupGroupIdentifierCollection = (Collection<?>) signupGroupIdentifiersObject;
 
+          Double signupStartDouble = (Double) match.get("signupStart");
+          Double signupEndDouble = (Double) match.get("signupEnd");
+
+          if (signupStartDouble != null) {
+            long itemLong = (long) (signupStartDouble * 1000);
+            Date signupStart = new Date(itemLong);
+            if (signupStart != null && !signupStart.equals(new Date())) {
```

Comment:
Unnecessary if clause. signupStart cannot be null at this point and since new Date() is to a millisecond, it is very unlikely to equal signupStart either.